src/util/selinux.c

```diff
@@ -43,25 +44,47 @@
  */
 int selinux_file_context(const char *dst_name)
 {
+    struct selabel_handle *handle = NULL;
     char *scontext = NULL;
+    int ret;
 
     if (is_selinux_enabled() == 1) {
         /* Get the default security context for this file */
-        if (matchpathcon(dst_name, 0, &scontext) < 0) {
-            if (security_getenforce () != 0) {
-                return 1;
-            }
+        handle = selabel_open(SELABEL_CTX_FILE, NULL, 0);
+        if (handle == NULL) {
+            ret = errno;
+            DEBUG(SSSDBG_CRIT_FAILURE, "Unable to create selabel context "
+                  "[%d]: %s\n", ret, sss_strerror(ret));
```

Comment:
Why do not use errno directly here?

Reply:
Because errno is an overused global variable. You never know if the value is going to be changed inside the macro it is safer to store the value first.